### PR TITLE
[cpu] Use port mask from static shape inference

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Use port mask defined in Static shape inference classes used for shape inference defined in core
 - Apply use this port mask in CPU nodes which are using `NgraphShapeInferFactory` class.

### Tickets:
 - CVS-118703 
 - CVS-119176

### CODE REVIEW

File: src/plugins/intel_cpu/src/shape_inference/shape_inference_ngraph.hpp [hunk 1]
```diff
@@ -11,14 +11,24 @@ namespace ov {
...
+     */
+    NgraphShapeInfer(std::shared_ptr<IStaticShapeInfer> shape_infer);
+
+    /**
+     * @brief Wraps IStaticShapeInfer into IShapeInfer interface. Will use port mask defined by user
+     *
+     * @param shape_infer Instance of IStaticShapeInfer.
+     * @param port_mask   Port mask define.
+     */
+    NgraphShapeInfer(std::shared_ptr<IStaticShapeInfer> shape_infer, IShapeInfer::port_mask_t port_mask);
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/shape_inference/shape_inference_ngraph.hpp:31]: Is it possible to remove this constructor and replace the  `get_port_mask()` with:
```cpp
    port_mask_t get_port_mask() const override {
        return m_shape_infer->get_port_mask();
    }
```
?

**Comment by tiger100256-hu** [comment]
@praasz  I have found that there are some redundancy code. for example https://github.com/openvinotoolkit/openvino/blob/2f4f76070faa921d3d0a720639e07d5310c50b6d/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L537
there are two line **_OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::BatchNormInference, ShapeInferBase)** and three line 
**_OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy)**
```
    _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::BatchNormInference, ShapeInferBase),
    _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::BatchNormInference, ShapeInferBase),
    _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy),
    _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy),
    _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy),
```
and also here
https://github.com/openvinotoolkit/openvino/blob/2f4f76070faa921d3d0a720639e07d5310c50b6d/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L432
I think one line is enough, could you help to remove others in this PR, thanks

**Comment by praasz** [comment]
> @praasz I have found that there are some redundancy code. for example
> 
> https://github.com/openvinotoolkit/openvino/blob/2f4f76070faa921d3d0a720639e07d5310c50b6d/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L537
> 
> 
> there are two line **_OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::BatchNormInference, ShapeInferBase)** and three line
> **_OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy)**
> ```
>     _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::BatchNormInference, ShapeInferBase),
>     _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::BatchNormInference, ShapeInferBase),
>     _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy),
>     _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy),
>     _OV_OP_SHAPE_INFER_NON_TEMPLATE_REG(opset1::Convert, ShapeInferCopy),
> ```
> 
> and also here
> 
> https://github.com/openvinotoolkit/openvino/blob/2f4f76070faa921d3d0a720639e07d5310c50b6d/src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp#L432
> 
> 
> I think one line is enough, could you help to remove others in this PR, thanks

Duplicates removed.

**Comment by maxnick** [comment]
@praasz, it think it does make sense to run the internal performance validation for dynamic shapes models, just to make sure that there are no functional degradations. Let me trigger the job.

**Review by maxnick — APPROVED** [review]
Internal validation passed successfully.

**Pull request merged**